guides/micronaut-mongodb-synchronous/java/src/main/java/example/micronaut/MongoDbFruitRepository.java

```diff
@@ -0,0 +1,40 @@
+package example.micronaut;
+
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import io.micronaut.core.annotation.NonNull;
+import jakarta.inject.Singleton;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+@Singleton // <1>
+public class MongoDbFruitRepository implements FruitRepository {
+
+    private final MongoDbConfiguration mongoConf;
+    private final MongoClient mongoClient;
+
+    public MongoDbFruitRepository(MongoDbConfiguration mongoConf,  // <2>
+                                  MongoClient mongoClient) {  // <3>
+        this.mongoConf = mongoConf;
+        this.mongoClient = mongoClient;
+    }
+
+    @Override
+    public void save(@NonNull @NotNull @Valid Fruit fruit){
+        getCollection().insertOne(fruit);
+    }
```

Comment:
Not sure if this is possible or not but I think it would be better if you add an `id` field to `Fruit` (probably an `UUID`) and configure the POJO (or Mongo) to assign it automatically.

Reply:
I would prefer to have different POJOs. I would like to use the same POJO for the controller layer and the persistence layer.

Reply:
I think we could do this as an improvement after the guide is published.